src/sentry/incidents/tasks.py

```diff
@@ -215,11 +219,14 @@ def process_pending_incident_snapshots():
             process_pending_incident_snapshots.apply_async(countdown=1)
             break
         else:
-            with transaction.atomic():
-                if (
-                    incident.status == IncidentStatus.CLOSED.value
-                    and not IncidentSnapshot.objects.filter(incident=incident).exists()
-                ):
-                    if IncidentProject.objects.filter(incident=incident).exists():
-                        create_incident_snapshot(incident, windowed_stats=True)
-                pending_snapshot.delete()
+            try:
+                with transaction.atomic():
+                    if (
+                        incident.status == IncidentStatus.CLOSED.value
+                        and not IncidentSnapshot.objects.filter(incident=incident).exists()
+                    ):
+                        if IncidentProject.objects.filter(incident=incident).exists():
+                            create_incident_snapshot(incident, windowed_stats=True)
+                    pending_snapshot.delete()
+            except Exception:
+                logger.exception("An error occurred while taking an incident snapshot")
```

Comment:
Do failed snapshots get retried?

Reply:
Yeah, they're rows in `PendingIncidentSnapshot` and will retry forever basically, until the row is deleted. So we can still fix any issues we see.